sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/control_flow/do_while/invalid_pipeline.yml

```diff
@@ -23,6 +23,25 @@ settings:
   force_rerun: true
 
 jobs:
+  invalid_condition_body:
+    type: pipeline
+    inputs:
+      job_in_number: ${{parent.inputs.job_in_number}}
+      job_in_number_1: ${{parent.inputs.job_in_number}}
+      job_in_path: ${{parent.inputs.job_in_path}}
+    component: ./components/do_while_body_pipeline_component.yml
+  invalid_condition:
+    type: do_while
+    body: ${{parent.jobs.invalid_condition_body}}
+    limits:
+      max_iteration_count: 5
+    condition: false
```

Comment:
how do we know false is a bool or a str?

Reply:
`yaml.safe_load` handles this, `false` will be parsed as bool and `"false"` will be parsed as string.

Reply:
Remove support for bool type, still raise validation error on type.